kolibri/plugins/facility/assets/src/views/ManageClassPage/index.vue

```diff
@@ -120,17 +204,107 @@
       FacilityAppBarPage,
       ClassCreateModal,
       ClassDeleteModal,
+      ClassRenameModal,
+      SidePanelModal,
+      SelectableList,
     },
     mixins: [commonCoreStrings],
     setup() {
+      const classDetails = ref({
+        id: '',
+        name: '',
+      });
+      const classCoachesIds = ref([]);
+      const classCoaches = ref([]);
+      const facilityUsers = ref([]);
+      const copiedClassName = ref(null);
+      const openCopyClassPanel = ref(false);
+      const openRenameModal = ref(false);
       const { classToDelete, selectClassToDelete, clearClassToDelete } = useDeleteClass();
       const { getFacilities, userIsMultiFacilityAdmin } = useFacilities();
+      const { createSnackbar } = useSnackbar();
+      const { $store, $router } = getCurrentInstance().proxy;
+      const activeFacilityId =
+        $router.currentRoute.params.facility_id || $store.getters.activeFacilityId;
+
+      const {
+        copyClasslabel$,
+        renameClassLabel$,
+        coachesAssignedToClassLabel$,
+        classTitleLabel$,
+        selectAllLabel$,
+        numCoachesSelected$,
+        classCopiedSuccessfully$,
+        copyOfClass$,
+        classNameAlreadyExists$,
+      } = bulkUserManagementStrings;
+
+      const handleSelection = newSelection => {
+        classCoachesIds.value = newSelection;
+      };
+
+      const fetchFacilityUsers = async () => {
+        const resp = await FacilityUserResource.fetchCollection({
+          getParams: pickBy({
+            member_of: activeFacilityId,
+            user_type: UserKinds.COACH,
+          }),
+          force: true,
+        });
+        facilityUsers.value = resp;
+      };
+
+      const handleOptionSelection = (selection, row) => {
+        classDetails.value = row;
+        copiedClassName.value = copyOfClass$({ class: classDetails.value.name });
+
+        classCoachesIds.value = classDetails.value.coaches.map(coach => coach.id);
+        classCoaches.value = facilityUsers.value.map(coach => ({
+          id: coach.id,
+          username: coach.username,
+          full_name: coach.full_name,
+          label: coach.full_name,
+        }));
+
+        if (selection.value === Modals.DELETE_CLASS) {
+          selectClassToDelete(row);
+          return;
+        }
+
+        if (selection.value === Modals.EDIT_CLASS_NAME) {
+          openRenameModal.value = true;
+          return;
+        }
+
+        if (selection.value === Modals.COPY_CLASS) {
+          openCopyClassPanel.value = true;
+          return;
+        }
+      };
+
+      fetchFacilityUsers();
```

Comment:
Just a heads up that we are loading these coaches on every ManageClassPage render, even if we dont want to copy the class. Moving the class copy to its own component will prevent this.